src/components/Personalization/turbine/executeRules.js

```diff
@@ -39,17 +39,22 @@ export default (rules, ruleComponentModules, logger) => {
   };
 
   const getErrorMessage = (ruleComponent, rule, errorMessage, errorStack) => {
-    return `Failed to execute ${ruleComponent.moduleType} for ${
-      rule.name
-    } rule. ${errorMessage} ${errorStack ? `\n ${errorStack}` : ""}`;
+    const ruleRespresentation = JSON.stringify(rule);
+
+    return `Failed to execute ${
+      ruleComponent.moduleType
+    } for ${ruleRespresentation} rule.
+    ${errorMessage} ${errorStack ? `\n ${errorStack}` : ""}`;
   };
 
   const logActionError = (action, rule, e) => {
     logger.error(getErrorMessage(action, rule, e.message, e.stack));
   };
 
   const logRuleCompleted = rule => {
-    logger.log(`Rule ${rule.name} fired.`);
+    const ruleRespresentation = JSON.stringify(rule);
```

Comment:
Can you wrap this in a `if (logger.enabled) { ... }`?

Stringify can be expensive especially if the rule is large.  We should only call it if the logger is enabled.

Reply:
yes, sure, thank you